SerialPrograms/Source/PokemonSV/Programs/Sandwiches/PokemonSV_SandwichMaker.cpp

```diff
@@ -0,0 +1,507 @@
+/*  Sandwich Maker
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/Notifications/ProgramNotifications.h"
+#include "CommonFramework/Exceptions/OperationFailedException.h"
+#include "CommonFramework/Tools/VideoResolutionCheck.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "CommonFramework/InferenceInfra/InferenceRoutines.h"
+#include "Pokemon/Pokemon_Strings.h"
+#include "Common/Cpp/Exceptions.h"
+#include "CommonFramework/ImageTools/ImageFilter.h"
+#include "PokemonSV/Programs/Sandwiches/PokemonSV_SandwichRoutines.h"
+#include "PokemonSV/Resources/PokemonSV_Ingredients.h"
+#include "PokemonSV/Resources/PokemonSV_FillingsCoordinates.h"
+#include "PokemonSV/Programs/Sandwiches/PokemonSV_IngredientSession.h"
+#include "PokemonSV/Inference/Picnics/PokemonSV_SandwichHandDetector.h"
+#include "PokemonSV/Inference/Picnics/PokemonSV_SandwichIngredientDetector.h"
+#include "PokemonSV/Inference/Picnics/PokemonSV_SandwichRecipeDetector.h"
+#include "PokemonSV_SandwichMaker.h"
+
+//#include <iostream>
+//using std::cout;
+//using std::endl;
+#include <unordered_map>
+#include <algorithm>
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+using namespace Pokemon;
+
+SandwichMaker_Descriptor::SandwichMaker_Descriptor()
+    : SingleSwitchProgramDescriptor(
+        "PokemonSV:SandwichMaker",
+        STRING_POKEMON + " SV", "Sandwich Maker",
+        "ComputerControl/blob/master/Wiki/Programs/PokemonSV/SandwichMaker.md",
+        "Make a sandwich of your choice.",
+        FeedbackType::REQUIRED,
+        AllowCommandsWhenRunning::DISABLE_COMMANDS,
+        PABotBaseLevel::PABOTBASE_12KB
+    )
+{}
+
+SandwichMaker::SandwichMaker()
+    : GO_HOME_WHEN_DONE(false)
+    , NOTIFICATIONS({
+        &NOTIFICATION_PROGRAM_FINISH,
+        &NOTIFICATION_ERROR_FATAL,
+        })
+{
+    PA_ADD_OPTION(SANDWICH_OPTIONS);
+    PA_ADD_OPTION(GO_HOME_WHEN_DONE);
+    PA_ADD_OPTION(NOTIFICATIONS);
+}
+
+void SandwichMaker::program(SingleSwitchProgramEnvironment& env, BotBaseContext& context) {
```

Comment:
Haven't looked through the logic here yet, but it looks like the building logic is here. Since we intend to build sandwiches in other programs (like the platform bot), can we move all of this into a separate function instead of inlining it into the sandwich builder program?